wgpu-hal/src/dx12/device.rs

```diff
@@ -956,13 +982,24 @@ impl crate::Device for super::Device {
                     _ => todo!(),
                 };
 
+                let binding_array_size = entry.count.map(NonZeroU32::get);
```

Comment:
nit: binding arrays and dynamic buffers are mutually exclusive

Reply:
Should I update this to not use the `binding_array_size` and assert instead? Or just ignore it with a comment?